### ISSUE TITLE
Vertex winding order

### ISSUE THREAD

**Issue by davenquinn** [mapbox/martini#11]
I'm working on [a prototype module](https://github.com/davenquinn/cesium-martini/blob/master/src/terrain-provider.ts) that uses this library + Mapbox elevation tiles to render [`QuantizedMeshTerrainData`](https://cesium.com/docs/cesiumjs-ref-doc/QuantizedMeshTerrainData.html) tiles in Cesium.

I've managed to get a version working that creates a mesh in Cesium, but all of the tiles are (rather hilariously) warped.

<img width="1008" alt="image" src="https://user-images.githubusercontent.com/1537910/80405311-0b9d2800-8888-11ea-91b3-2cfb651cc924.png">

I think the issue relates to some combination of the winding order of triangles and the sorting of vertices. Cesium apparently expects triangles wound in counterclockwise order; there may be some restrictions on the sorting of vertices as well (see e.g. [this example dataset](https://github.com/CesiumGS/cesium/pull/778/files#r4383221)). I've tried several combinations of rewinding triangles and [sorting vertices](https://github.com/davenquinn/cesium-martini/blob/a9c06768b103c52cd419dfe7c4d6661d2bd32006/src/terrain-provider.ts#L193) to no avail.

This may be a better issue to raise in the Cesium context, but I thought I would start here: are there any guarantees on winding order given by this library? Does the pattern of artifacts in my image above look diagnostic of any particular failure mode? Perhaps a few words in the documentation about how the output mesh is structured would allay my confusion.

**Comment by mourner** [comment]
This seems like it could be the same issue as https://github.com/mapbox/martini/issues/5#issuecomment-527322212 — can you check?

**Comment by davenquinn** [comment]
Hey @mourner. Thanks for looking at this!

It works now! Thanks for resetting my brain. I had already looked at #5 and dismissed (I was already extracting heightfield values). But reviewing it again at your behest, I found an off-by-one error in indexing in my code. 🤦 

<img width="1008" alt="image" src="https://user-images.githubusercontent.com/1537910/80416601-03e67f00-889a-11ea-9c60-80af3a8c52e1.png">

I'll be working on some of the low-hanging performance and optimization fruit, and bundling into a module, but we can now represent global terrain in Cesium using solely streaming Mapbox data! Thanks for laying the groundwork for this advance.
